src/Transport/ApiSnmpClient.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SimPod\PhpSnmp\Transport;
+
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestFactoryInterface;
+use Psr\Http\Message\StreamFactoryInterface;
+use SimPod\PhpSnmp\Exception\EndOfMibReached;
+use SimPod\PhpSnmp\Exception\GeneralException;
+use SimPod\PhpSnmp\Exception\NoSuchInstanceExists;
+use SimPod\PhpSnmp\Exception\NoSuchObjectExists;
+use Throwable;
+use function array_key_exists;
+use function assert;
+use function count;
+use function is_string;
+use function Safe\json_decode;
+use function Safe\json_encode;
+use function Safe\preg_match;
+use const JSON_BIGINT_AS_STRING;
+
+final class ApiSnmpClient implements SnmpClient
+{
+    private const API_PATH = '/snmp-proxy';
+
+    private const REQUEST_TYPE_GET      = 'get';
+    private const REQUEST_TYPE_GET_NEXT = 'getNext';
+    private const REQUEST_TYPE_WALK     = 'walk';
+
+    /** @var ClientInterface */
+    private $client;
+
+    /** @var RequestFactoryInterface */
+    private $requestFactory;
+
+    /** @var StreamFactoryInterface */
+    private $streamFactory;
+
+    /** @var string */
+    private $apiHostUrl;
+
+    /** @var string */
+    private $host;
+
+    /** @var string */
+    private $community;
+
+    /** @var int */
+    private $timeout;
+
+    /** @var int */
+    private $retries;
+
+    /** @var string */
+    private $version;
+
+    public function __construct(
+        ClientInterface $client,
+        RequestFactoryInterface $requestFactory,
+        StreamFactoryInterface $streamFactory,
+        string $apiHostUrl,
+        string $host = '127.0.0.1',
+        string $community = 'public',
+        int $timeout = 1,
+        int $retries = 3,
+        string $version = '2c'
+    ) {
+        $this->client         = $client;
+        $this->requestFactory = $requestFactory;
+        $this->streamFactory  = $streamFactory;
+        $this->apiHostUrl     = $apiHostUrl;
+        $this->host           = $host;
+        $this->community      = $community;
+        $this->timeout        = $timeout;
+        $this->retries        = $retries;
+        $this->version        = $version;
+    }
+
+    /** @inheritDoc */
+    public function get(array $oids) : array
+    {
+        return $this->executeRequest(self::REQUEST_TYPE_GET, $oids);
+    }
+
+    /** @inheritDoc */
+    public function getNext(array $oids) : array
+    {
+        return $this->executeRequest(self::REQUEST_TYPE_GET_NEXT, $oids);
+    }
+
+    /** @inheritDoc */
+    public function walk(string $oid) : array
+    {
+        return $this->executeRequest(self::REQUEST_TYPE_WALK, [$oid]);
+    }
+
+    /**
+     * @param string[] $oids
+     *
+     * @return array<string, mixed>
+     */
+    private function executeRequest(string $requestType, array $oids) : array
```

Comment:
`@param ApiSnmpClient::REQUEST_TYPE_GET | ApiSnmpClient::REQUEST_TYPE_GET_NEXT | ApiSnmpClient::REQUEST_TYPE_WALK $requestType`

Reply:
I don't see a need for that really

Reply:
type safety